src/tests/suite/daos_degrade_ec.c

```diff
@@ -161,6 +161,251 @@ degrade_full_fail_data_parity(void **state)
 	degrade_ec_internal(state, shards, 2, FULL_UPDATE);
 }
 
+static void
+degrade_dfs_internal(void **state, int *shards, int shards_nr)
+{
+	dfs_t		*dfs_mt;
+	daos_handle_t	co_hdl;
+	test_arg_t	*arg = *state;
+	d_sg_list_t	sgl;
+	d_iov_t		iov;
+	dfs_obj_t	*obj;
+	daos_size_t	buf_size = 32 * 1024 * 32;
+	daos_size_t	partial_size = 32 * 1024 * 2;
+	daos_size_t	chunk_size = 32 * 1024 * 4;
+	daos_size_t	fetch_size = 0;
+	uuid_t		co_uuid;
+	char		filename[32];
+	d_rank_t	ranks[4] = { -1 };
+	int		idx = 0;
+	daos_obj_id_t	oid;
+	char		*buf;
+	char		*vbuf;
+	int		i;
+	int		rc;
+
+	uuid_generate(co_uuid);
+	rc = dfs_cont_create(arg->pool.poh, co_uuid, NULL, &co_hdl,
+			     &dfs_mt);
+	assert_int_equal(rc, 0);
+	printf("Created DFS Container "DF_UUIDF"\n", DP_UUID(co_uuid));
+
+	D_ALLOC(buf, buf_size);
+	assert_true(buf != NULL);
+	D_ALLOC(vbuf, buf_size);
+	assert_true(vbuf != NULL);
+
+	d_iov_set(&iov, buf, buf_size);
+	sgl.sg_nr = 1;
+	sgl.sg_nr_out = 1;
+	sgl.sg_iovs = &iov;
+
+	dts_buf_render(buf, buf_size);
+	memcpy(vbuf, buf, buf_size);
+
+	/* Full stripe update */
+	sprintf(filename, "degrade_file");
+	rc = dfs_open(dfs_mt, NULL, filename, S_IFREG | S_IWUSR | S_IRUSR,
+		      O_RDWR | O_CREAT, DAOS_OC_EC_K4P2_L32K, chunk_size,
+		      NULL, &obj);
+	assert_int_equal(rc, 0);
+	rc = dfs_write(dfs_mt, obj, &sgl, 0, NULL);
+	assert_int_equal(rc, 0);
+
+	/* Partial update after that */
+	d_iov_set(&iov, buf, partial_size);
+	for (i = 0; i < 10; i++) {
+		dfs_write(dfs_mt, obj, &sgl, buf_size + i * 100 * 1024, NULL);
+		assert_int_equal(rc, 0);
+	}
+
+	dfs_obj2id(obj, &oid);
+	while (shards_nr-- > 0) {
+		ranks[idx] = get_rank_by_oid_shard(arg, oid, shards[idx]);
+		idx++;
+	}
+	rebuild_pools_ranks(&arg, 1, ranks, idx, false);
+
+	/* Verify full stripe */
+	d_iov_set(&iov, buf, buf_size);
+	fetch_size = 0;
+	rc = dfs_read(dfs_mt, obj, &sgl, 0, &fetch_size, NULL);
+	assert_int_equal(rc, 0);
+	assert_int_equal(fetch_size, buf_size);
+	assert_memory_equal(buf, vbuf, buf_size);
+
+	/* Verify partial stripe */
+	d_iov_set(&iov, buf, partial_size);
+	for (i = 0; i < 10; i++) {
+		memset(buf, 0, buf_size);
+		fetch_size = 0;
+		dfs_read(dfs_mt, obj, &sgl, buf_size + i * 100 * 1024,
+			 &fetch_size, NULL);
+		assert_int_equal(rc, 0);
+		assert_int_equal(fetch_size, partial_size);
+		assert_memory_equal(buf, vbuf, partial_size);
+	}
+
+	rc = dfs_release(obj);
+	assert_int_equal(rc, 0);
+
+	D_FREE(buf);
+	rc = dfs_umount(dfs_mt);
+ 	assert_int_equal(rc, 0);
+
+	rc = daos_cont_close(co_hdl, NULL);
+ 	assert_int_equal(rc, 0);
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs